### PR TITLE
Disable debugprint and optimization CLI by default

### PR DESCRIPTION
**Pull request by tbennun**

### CODE REVIEW

File: dace/config_schema.yml [hunk 1]
```diff
@@ -29,14 +29,22 @@ required:
                     Automatically specialize every SDFG to the symbol values 
                     at call-time. Requires all symbols to be set.
 
-            interface:
+            autooptimize:
```
Comment (definelicht) [inline dace/config_schema.yml:32]: Isn't this misleading? Auto-optimize sounds like DaCe will just make your program fast. Maybe just "optimizer" or "cli_optimizer" or...

File: dace/sdfg/state.py [hunk 2]
```diff
@@ -1526,6 +1526,9 @@ def add_memlet_path(self,
                     if propagate:
                         cur_memlet = propagate_memlet(self, cur_memlet, snode,
                                                       True)
+        # Try to initialize memlets
+        for edge in edges:
+            edge.data.try_initialize(self.parent, self, edge)
```
Comment (definelicht) [inline dace/sdfg/state.py:1531]: Is this related or did it sneak in?

Comment (tbennun) [reply dace/sdfg/state.py:1531]: It's related. The optimizer CLI, even when you don't optimize, actually serialized and deserialized, fixing the memlets. Now we just fix them during creation

File: dace/config_schema.yml (hunk 1 above)
Comment (tbennun) [reply dace/config_schema.yml:32]: You can still choose the optimizer in the next configuration to not be CLI (we use this in the transformation tester), so "cli_optimizer" is out. can we call it "optimize_on_call"?

Comment (definelicht) [reply dace/config_schema.yml:32]: Maybe something with "transform" is also more appropriate than "optimize"?

Comment (tbennun) [reply dace/config_schema.yml]: Done

Comment (definelicht) [reply dace/config_schema.yml]: I still don't like that it's called "autooptimize" for the above reason :spooderman:

Comment (tbennun) [reply dace/config_schema.yml]: But it's now called transform_on_call

**Comment by tbennun** [comment]
@definelicht I renamed it
